gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinServerIntegrateTest.java

```diff
@@ -349,6 +368,28 @@ private void shouldStoreUserAgentInContext() throws InterruptedException {
         }
     }
 
+    private void shouldStoreUserAgentInHttpContext() throws InterruptedException {
+        if(server.getChannelizer() instanceof TestChannelizer) {
+            TestChannelizer channelizer = (TestChannelizer) server.getChannelizer();
+            channelizer.resetChannelHandlerContext();
+            assertNull(getUserAgentIfAvailable());
+            final Cluster cluster = TestClientFactory.build()
+                    .channelizer(org.apache.tinkerpop.gremlin.driver.Channelizer.HttpChannelizer.class)
+                    .enableUserAgentOnConnect(true)
+                    .create();
+            final Client client = cluster.connect();
+
+            client.submit("g.V()");
+            java.lang.Thread.sleep(2000);
+            assertEquals(UserAgent.USER_AGENT, getUserAgentIfAvailable());
+            client.submit("g.V()");
+            java.lang.Thread.sleep(2000);
```

Comment:
Test adds 4 seconds to total test execution time.
Can you replace this condition with checking userAgent every 50-100ms, but no longer than 2 seconds?